index.js

```diff
@@ -27,7 +39,76 @@ app.use(express.json());
 // IDEAS METHODS
 
 app.get('/ideas', (req, res) => {
-    res.status(200).json(getIdeas());
+    let sortedIdeasList;
+
+    if (req.query.sortBy === 'all') {
+        sortedIdeasList = getIdeas();
+    } else if (req.query.sortBy === 'totalDonationSum') {
+        sortedIdeasList = sortedByDonationSumIdeas();
+    } else if (
+        req.query.sortBy === 'status' &&
+        req.query.status === 'accepted'
+    ) {
+        sortedIdeasList = approvedIdeasList();
+    } else if (
+        req.query.sortBy === 'status' &&
+        req.query.status === 'pending'
+    ) {
+        sortedIdeasList = pendingIdeasList();
+    } else if (
+        req.query.sortBy === 'status' &&
+        req.query.status === 'rejected'
+    ) {
+        sortedIdeasList = rejectedIdeasList();
+    }
+
+    try {
+        res.status(200).json(sortedIdeasList);
```

Comment:
This code is unlikely to throw an error, as you're not calling any functions of your own - you're calling Express API which is well tested. So this `try/catch` is not useful here

Reply:
Updated and tested.